server/post_processor.go

```diff
@@ -38,9 +38,11 @@ type PostProcessor struct {
 	excludeDirectMessages  bool
 	excludePrivateChannels bool
 
-	resultsCache *moderationResultsCache
-	postsCh      chan *model.Post
-	done         chan struct{}
+	resultsCache  *moderationResultsCache
+	postCache     *postCache
+	postsCh       chan *model.Post
+	done          chan struct{}
+	cleanupTicker *time.Ticker
```

Comment:
Any particular reason the cache cleanup logic is performed in the `PostProcessor` instead of in the `postCache` ?

Reply:
Thinking about this more - I'm using more goroutines than I need to. Each processor has two, but we really only need one each. I made the change in a74f8a0 and b61aad9 to drop the number of goroutines. With this change, I don't want to push cache cleanup into the caches, as that would require additional goroutines.

(The 6c2b52f commit was an incomplete solution of a74f8a0 that I force pushed over - can safely ignore that :sweat_smile:).